config/config.go

```diff
@@ -206,7 +227,11 @@ func (cfg *Config) NewNode(ctx context.Context) (host.Host, error) {
 		}
 
 		if hop {
-			h = relay.NewRelayHost(swrm.Context(), h.(*bhost.BasicHost), discovery)
+			// advertise ourselves
+			// TODO: Why do we only do this when EnableAutoRelay is
+			// set? This has absolutely _nothing_ to do with
+			// autorelay.
```

Comment:
@vyzo?

Reply:
Sure it has to do with autorelay -- it's the advertisement of relay services for autorelayed hosts!

Reply:
If you look at this from an "I'm enabling the autorelay subsystem" perspective, yes. However, users will just see an autorelay _option_. When looking at these as _options_,

1. If I enable relay, I expect my node to make use of relays and accept relay connections.
2. If I enable relay _hop_, I expect my node to _act_ as a relay. That _includes_ telling other users that I'm willing to be a relay.
3. If I enable autorelay, I expect my node to automatically pick and advertise a relay.

Having to enable both relay hop + autorelay to get other users to use me as a relay is really not what I would expect.

Reply:
We do need a mechanism to allow users to enable relay for their own purposes (ie their own little fleet of nodes) but _without_ having to be inundated by the torrent of traffic that comes when announcing the relay functionality.

Reply:
Sure, but that should really be an option on the relay service. I'll try to document all this to be less confusing but I know we're still going to be confusing users.